transport-classes-epoll/src/main/java/io/netty5/channel/epoll/NativeDatagramPacketArray.java

```diff
@@ -112,21 +110,24 @@ boolean addReadable(Buffer buf, int segmentLen, InetSocketAddress recipient) {
                 int writableBytes = c.readableBytes();
                 int byteCount = segmentLen == 0? writableBytes : Math.min(writableBytes, segmentLen);
                 if (iovArray.addReadable(c, byteCount)) {
-                    NativeDatagramPacket p = packets[count];
-                    long packetAddr = iovArray.memoryAddress(iovArrayStart);
-                    p.init(packetAddr, iovArray.count() - iovArrayStart, segmentLen, recipient);
-                    count++;
                     c.skipReadableBytes(byteCount);
+                } else {
+                    break;
```

Comment:
@chrisvest this was a bug that pre-existed before this PR.If you prefer I can also fix this in an extra PR.